### PR TITLE
Add icon and metadata to Windows Crystal compiler binary

### PR DESCRIPTION
**Pull request by HertzDevil**
This PR sets up generation of a Windows resource file and embeds it into the Crystal compiler binary, giving us some nice metadata:

image

image

The information is pulled straight from the Makefile. This is important because we cannot simply pull the data from `bin\crystal --version` before building a fresh compiler with the updated information. However, if we are going to do this with `shards.exe` as well, or even anything built on Windows, then perhaps compiler support is necessary.

The `1` after the patch number has the same meaning as it does in Crystal's release packages (e.g. `crystal-1.5.1-1`. It is not customizable yet.

The icon shares the same file with the playground's favicon. It has been updated to include a 256×256 subimage since [it is the largest size that Windows supports](https://learn.microsoft.com/en-us/windows/apps/design/style/iconography/app-icon-construction#icon-scaling).

### CODE REVIEW

File: Makefile.win [hunk 1]
```diff
@@ -199,6 +208,49 @@ clean_crystal: ## Clean up crystal built files
 clean_cache: ## Clean up CRYSTAL_CACHE_DIR files
 	$(call RMDIR,"$(shell .\bin\crystal env CRYSTAL_CACHE_DIR)")
 
+.PHONY: rc
+rc: ## Write compiler resource script to standard output
+	$(eval comma := ,)
+	$(eval ver_comma := $(subst .,$(comma),$(subst -dev,,$(CRYSTAL_VERSION))))
+	$(eval build_date := $(shell git show -s --format=%cs HEAD))
+	$(eval build_year := $(firstword $(subst -, ,$(build_date))))
```
Comment (straight-shoota) [inline Makefile.win:216]: I'd suggest to name these variables `source_date` and `source_year`. They do not represent the date of the build.
Also support for `SOURCE_DATE_EPOCH` would be nice (which is already used in this Makefile for the source date reported by the compiler itself).

File: Makefile.win [hunk 2]
```diff
@@ -199,6 +208,49 @@ clean_crystal: ## Clean up crystal built files
...
+	echo     FILETYPE       VFT_APP&\
+	echo     FILESUBTYPE    VFT2_UNKNOWN&\
+	echo BEGIN&\
+	echo     BLOCK "StringFileInfo"&\
+	echo     BEGIN&\
+	echo         BLOCK "040904B0"&\
+	echo         BEGIN&\
+	echo             VALUE "ProductName",      "Crystal"&\
+	echo             VALUE "ProductVersion",   "$(ver_full)"&\
+	echo             VALUE "FileVersion",      "$(ver_full)"&\
```
Comment (straight-shoota) [inline Makefile.win:239]: Is this correct? File version in the screenshot shows `1.6.0.1`, not the value of `ver_full`.

**Comment by Sija** [comment]
> The icon shares the same file with the playground's favicon. It has been updated to include a 256×256 subimage since [it is the largest size that Windows supports](https://learn.microsoft.com/en-us/windows/apps/design/style/iconography/app-icon-construction#icon-scaling).

@HertzDevil I'd suggest using a separate icon file instead of downsizing the playground icon.

**Comment by HertzDevil** [comment]
The favicon was originally at 128x128

File: Makefile.win (hunk 2 above)
Comment (HertzDevil) [reply Makefile.win:239]: Yeah this is correct, the dialog just displays the `FILEVERSION` field above but it looks like this field here is also required.

File: Makefile.win (hunk 1 above)
Comment (HertzDevil) [reply Makefile.win:216]: Conversion from a Unix timestamp is not possible with CMD nor with the built-in version of PowerShell (`Get-Date` on version 5.1 does not support the `-UnixTimeSeconds` parameter).

Comment (HertzDevil) [reply Makefile.win]: One evil approach is:

```
$(eval source_date := $(shell crystal eval puts Time.unix($(SOURCE_DATE_EPOCH)).to_s(%(%Y-%m-%d))))
```

A more evil one is to run the following _Crystal_ code with the empty prelude inside the Makefile:

```crystal
lib LibC
  struct Tm
    tm_sec : Int32
    tm_min : Int32
    tm_hour : Int32
    tm_mday : Int32
    tm_mon : Int32
    tm_year : Int32
    tm_wday : Int32
    tm_yday : Int32
    tm_isdst : Int32
  end

  fun _gmtime64(timer : Int64*) : Tm*
  fun strftime(str : UInt8*, count : UInt64, format : UInt8*, time : Tm*) : UInt64
  fun printf(format : UInt8*, ...) : Int32
end

struct StaticArray(T, N)
  def to_unsafe
    pointerof(@buffer)
  end
end

class String
  def to_unsafe
    pointerof(@c)
  end
end

t = {{ env("SOURCE_DATE_EPOCH").to_i }}.to_i64!
tm = LibC._gmtime64(pointerof(t))
buf = uninitialized UInt8[11]
LibC.strftime(buf, 11, "%Y-%m-%d", tm)
LibC.printf("%s", buf)
```

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**